src/files-regular/add-pull-stream.js

```diff
@@ -44,35 +45,21 @@ module.exports = (common, options) => {
 
       const stream = ipfs.addPullStream()
 
-      pull(
-        pull.values(files),
-        stream,
-        pull.collect((err, filesAdded) => {
-          expect(err).to.not.exist()
+      const filesAdded = await pullToPromise.any(pull(pull.values(files), stream))
+      const testFolderIndex = filesAdded.length - 1
 
-          filesAdded.forEach((file) => {
-            if (file.path === 'test-folder') {
-              expect(file.hash).to.equal(fixtures.directory.cid)
-              done()
-            }
-          })
-        })
-      )
+      expect(filesAdded).to.have.nested.property(`[${testFolderIndex}].path`, 'test-folder')
+      expect(filesAdded).to.have.nested.property(`[${testFolderIndex}].hash`, fixtures.directory.cid)
     })
 
-    it('should add with object chunks and pull stream content', (done) => {
+    it('should add with object chunks and pull stream content', async () => {
       const expectedCid = 'QmRf22bZar3WKmojipms22PkXH1MZGmvsqzQtuSvQE3uhm'
+      const data = [{ content: pull.values([Buffer.from('test')]) }]
+      const stream = ipfs.addPullStream()
 
-      pull(
-        pull.values([{ content: pull.values([Buffer.from('test')]) }]),
-        ipfs.addPullStream(),
-        pull.collect((err, res) => {
-          if (err) return done(err)
-          expect(res).to.have.length(1)
-          expect(res[0]).to.deep.equal({ path: expectedCid, hash: expectedCid, size: 12 })
-          done()
-        })
-      )
+      const res = await pullToPromise.any(pull(pull.values(data), stream))
```

Comment:
IMO pull-streams pipelines should not be transformed in promises, you lose flexibility and gain 0 value. Pull-streams should either disappear in the future or be left to work as they were created.